src/map/clif.c

```diff
@@ -19136,6 +19214,104 @@ void clif_parse_sale_remove( int fd, struct map_session_data* sd ){
 #endif
 }
 
+/// Achievement System
+/// Author: Luxuri, Aleos
+
+/**
+ * Sends all achievement data to the client (ZC_ALL_AG_LIST).
+ * 0a23 <packetType>.W <packetLength>.W <ACHCount>.L <ACHPoint>.L
+ */
+void clif_achievement_list_all(struct map_session_data *sd)
+{
+	int i, j, len, fd, *info;
+	uint16 count = 0;
+
+	nullpo_retv(sd);
+
+	fd = sd->fd;
+	count = sd->achievement_data.count; // All achievements should be sent to the client
+	len = (50 * count) + 22;
+
+	if (len <= 22)
+		return;
+
+	info = achievement_level(sd, true);
+
+	WFIFOHEAD(fd,len);
+	WFIFOW(fd, 0) = 0xa23;
+	WFIFOW(fd, 2) = len;
+	WFIFOL(fd, 4) = count; // Cannot tell what this is? But it's important, either way!
+	WFIFOL(fd, 8) = sd->achievement_data.total_score; // Top number
+	WFIFOW(fd, 12) = sd->achievement_data.level; // Achievement Level (gold circle)
+	WFIFOL(fd, 14) = info[0]; // Achievements done (left number in bar)
+	WFIFOL(fd, 18) = info[1]; // Achievements TNL (right number in bar)
```

Comment:
Same here. The info variable contains the correct information, it's just the comments that are a bit misleading. Will update all these. Thanks!